consopt.go

```diff
@@ -232,3 +236,63 @@ func WithWeights(w *G.Node) ConsOpt {
 		return layer, nil
 	}
 }
+
+// WithKernelShape sets the kernel shape of the layer
+func WithKernelShape(s tensor.Shape) ConsOpt {
+	return func(l Layer) (Layer, error) {
+		switch c := l.(type) {
+		case *Conv:
+			c.kernelShape = s
+
+			return c, nil
+		case *MaxPool:
+			c.kernelShape = s
+
+			return c, nil
+		}
+
+		return nil, fmt.Errorf("Setting kernel shape is not supported by this layer: %T", l)
+	}
+}
+
+// WithPad sets the pad of the layer
+func WithPad(p []int) ConsOpt {
+	return func(l Layer) (Layer, error) {
+		switch c := l.(type) {
+		case *Conv:
+			c.pad = p
+
+			return c, nil
+		}
+
+		return nil, fmt.Errorf("Setting pad is not supported by this layer")
+	}
+}
+
+// WithStride sets the stride of the layer
```

Comment:
Might want to leave a comment saying this is for the convolution layers

Reply:
I'll do that